### DISCUSSION TITLE
Trying to match messages from grib with csv documentation

### DISCUSSION THREAD

**Discussion by JesseCrocker** [NOAA-GSL/GSL-Regional-Model-Forum#28] [Q&A] [answered]
Im attempting to match up messages from a grib file(currently testing with s3://noaa-hrrr-bdp-pds/hrrr.20220404/conus/hrrr.t00z.wrfsfcf00.grib2) with the entries in the CSV file at https://rapidrefresh.noaa.gov/hrrr/HRRRv4_GRIB2_WRFTWO.txt. I had initially assumed I could match them by record number, but while the first few records match, they do not all match. Is there any way to correlate between the two?

I am using pygrib to parse the grib files.

**Comment by blaylockbk** [comment, marked as answer]
Each GRIB2 file on AWS has a corresponding inventory file produced by wgrib2.

For example,
https://noaa-hrrr-bdp-pds.s3.amazonaws.com/hrrr.20220404/conus/hrrr.t00z.wrfsfcf00.grib2.idx

It doesn't have as much info as that .txt file you referred to, but the wgrib2 name could be matched between the two files.

**Reply by blaylockbk** [reply]
An alternative to pygrib if you haven't considered it is [cfgrib](https://github.com/ecmwf/cfgrib), which loads GRIB2 data into xarray using eccodes. The attributes of the loaded file contain many GRIB parameters. 

These are the attributes of the DataArray when I load 2-m temperature.
<img width="531" alt="image" src="https://user-images.githubusercontent.com/6249613/164906518-43fc299c-e965-4014-97db-2aeeac93d552.png">
<img width="511" alt="image" src="https://user-images.githubusercontent.com/6249613/164904001-7996aefc-216f-4518-a473-73419bffba3d.png">

(I retrieved and loaded this data from AWS with [Herbie](https://github.com/blaylockbk/Herbie))
